### PR TITLE
[#174] Build and publish Docker images via CircleCI

### PR DESCRIPTION
**Pull request by kamilongus**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+            - build
+          filters:
+            branches:
+              only: develop
+      - build_and_publish_docker_images:
+          requires:
+            - approve
+          filters:
+            branches:
+              only: develop
```
Comment (Som3) [inline .circleci/config.yml]: maybe need to push only master branch?

**Comment by kamilongus** [comment]
> seems legit to me, but i didn't find how to build/publish docker image via CircleCI
> Or how it will work?
Look at the CircleCI `workflows` tab - it has the `approve` task, but only for builds from the develop branch.

File: Makefile [hunk 2]
```diff
@@ -3,57 +3,56 @@
 # Run Postgres
 .PHONY: db
 db:
-  docker start codesearch-db || docker run --name codesearch-db \
```
Comment (neongreen) [inline Makefile]: If all makefiles have been switched to tabs, then you can remove the gmake requirement on macOS from the readme.

Comment (neongreen) [reply Makefile]: Oh, and then you can also remove RECIPEPREFIX.

File: Makefile [hunk 3]
```diff
@@ -3,57 +3,56 @@
...
+		-f "docker/$*/Dockerfile" \
+		-t "quay.io/aelve/codesearch-$*:local" .
+	docker tag \
+		"quay.io/aelve/codesearch-$*:local" \
+		"quay.io/aelve/codesearch-$*:latest"
 
 # Push a Docker image to Quay
-docker-push-%:
-  docker push "quay.io/aelve/codesearch-$*:latest"
+push-docker-%:
```
Comment (neongreen) [inline Makefile]: Don't forget to update readmes and the wiki (if needed) once this is merged. I think those commands were mentioned in the readmes somewhere, but I'm not sure.

File: Makefile (hunk 2 above)
Comment (kamilongus) [reply Makefile]: Done.

File: Makefile (hunk 3 above)
Comment (kamilongus) [reply Makefile]: Done.

File: .circleci/config.yml [hunk 4]
```diff
@@ -50,3 +63,34 @@ jobs:
...
+            - codesearch-core.jar
+            - codesearch-server.jar
+
+  build_and_publish_docker_images:
+    working_directory: ~/codesearch
+    machine:
+      docker_layer_caching: true
+    steps:
+      - run:
+          docker login "$REGISTRY" -u "$DOCKER_USER" -p "$DOCKER_PASS"
```
Comment (neongreen) [inline .circleci/config.yml]: There's no point in specifying $REGISTRY because we're locked into Quay anyway – it's in our Makefile.

**Comment by neongreen** [comment]
I'm going to be responsible for this PR. Once you've switched it to using `docker tag` (if you're going to do it at all), please ping for a re-review.

File: Makefile [hunk 5]
```diff
@@ -44,13 +46,35 @@ serve:
 
 # Build a Docker image (the project must be built already)
 build-docker-%:
```
Comment (neongreen) [inline Makefile]: It should still do something useful if `branch=` is not present. Building and tagging with `latest` seems adequate in this case.

**Pull request merged**